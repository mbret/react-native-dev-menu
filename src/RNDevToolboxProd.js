/**
 * @flow
 */
import React from 'react'
import { RNDevToolboxBase } from './RNDevToolboxBase'
import { RNDevToolboxContext } from './RNDevToolboxContext'
import type { RNDevToolboxProps, RNDevToolboxState } from './RNDevToolboxBase'

export class RNDevToolboxProd extends RNDevToolboxBase<RNDevToolboxProps<{}>, RNDevToolboxState<{}>> {
  state = {
    opened: false,
    actions: [],
    debug: null,
    indicators: [],
    tipsModalVisible: false
  }

  /**
   * We only render children inside context
   * to no broke the global api
   * @returns {function}
   */
  render () {
    return (
      <RNDevToolboxContext.Provider value={this}>
        {this.props.children}
      </RNDevToolboxContext.Provider>
    )
  }
}
