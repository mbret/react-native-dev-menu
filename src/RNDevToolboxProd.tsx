/* eslint no-unused-vars:0 */
/**
 *
 */
import React from 'react'
import { RNDevToolboxContext } from './RNDevToolboxContext'
import { RNDevToolboxProps, RNDevToolboxState, RNDevToolboxBase } from './RNDevToolboxBase'

export class RNDevToolboxProd extends RNDevToolboxBase<RNDevToolboxProps<{}>, RNDevToolboxState> {
  constructor (props: RNDevToolboxProps<{}>) {
    super(props)
    this.state = {
      opened: false,
      actions: [],
      debug: null,
      indicators: [],
      tipsModalVisible: false
    }
  }

  /**
   * We only render children inside context
   * to no broke the global api
   * @returns {function}
   */
  render () {
    return (
      <RNDevToolboxContext.Provider value={this}>
        {this.props.children}
      </RNDevToolboxContext.Provider>
    )
  }
}
