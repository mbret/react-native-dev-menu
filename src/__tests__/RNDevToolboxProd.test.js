/* eslint-env jest */
/**
 * @see https://medium.com/@jigishchawda/we-will-be-writing-unit-tests-for-the-above-mentioned-login-component-9e02002e7ff8
 * @flow
 */
import React from 'react'
import { RNDevToolboxProd } from '../RNDevToolboxProd'
import renderer from 'react-test-renderer'

describe('<RNDevToolboxProd>', () => {
  test('renders correctly', () => {
    const a = renderer.create(<RNDevToolboxProd />).toJSON()
    expect(a).toMatchSnapshot()

    const b = renderer.create(<RNDevToolboxProd onRef={() => {}} />).toJSON()
    expect(b).toMatchSnapshot()
  })

  test('.registerAction()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.registerAction({ name: 'foo', task: () => {} })).toBe(undefined)
  })

  test('.processAction()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.processAction('foo')).toBe(undefined)
  })

  test('.open()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.open()).toBe(undefined)
  })

  test('.close()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.close()).toBe(undefined)
  })

  test('.debug()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.debug()).toBe(undefined)
  })

  test('.toggle()', () => {
    const myComp = renderer.create(<RNDevToolboxProd />).getInstance()

    expect(myComp.toggle()).toBe(undefined)
    myComp.setState({ opened: true })
    expect(myComp.toggle()).toBe(undefined)
  })
})
