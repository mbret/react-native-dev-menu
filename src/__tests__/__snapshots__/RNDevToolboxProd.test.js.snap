// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RNDevToolboxProd> renders correctly 1`] = `null`;

exports[`<RNDevToolboxProd> renders correctly 2`] = `null`;
